watchdog/watcher/manager.go

```diff
@@ -20,45 +20,51 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-type Manager struct {
-	config    *config.Config
-	watchers  map[string]*Watcher
-	stopChans map[string]*chan bool
+type Manager interface {
+	Get(rsName string) *Watcher
+	HasWatcher(rsName string) bool
+	Watch(rs *replset.Replset)
 }
 
-func NewManager(config *config.Config) *Manager {
-	return &Manager{
-		config:    config,
-		watchers:  make(map[string]*Watcher),
-		stopChans: make(map[string]*chan bool),
+type WatcherManager struct {
```

Comment:
type name will be used as watcher.WatcherManager by other packages, and that stutters; consider calling this Manager